src/webOAuthServer.ts

```diff
@@ -189,6 +195,7 @@ export class WebOAuthServer extends AsyncCreatable<WebOAuthServer.Options> {
         this.logger.debug(`Successfully obtained auth code: ...${authCode.substring(authCode.length - 5)}`);
       } else {
         this.logger.debug('Expected an auth code but could not find one.');
+        throw SfdxError.create('@salesforce/core', 'auth', 'missingAuthCode');
```

Comment:
I'm not sure why we would ever continue without an authcode. You get `Value of undefined` if there isn't an auth code.